include/Vector.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+#include <iostream>
+#include <algorithm>
+
+class Vector {
+public:
+    Vector() = default;
+    Vector(size_t initial_size, int initial_value);
+    ~Vector();
+    Vector(const Vector& other);
+    Vector& operator=(const Vector& other);
+    void push_back(int value);
+    void pop_back();
+    int& operator[](size_t index);
```

Comment:
You are correct that the `operator[]` signature returns `T&`, but specifically for a trivially-copyable type like int there is no need to return a reference, the compiler will just omit it and copy the value